Source/BSN.Commons/Infrastructure/IDynamicFilterableRepository.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Linq.Expressions;
+
+namespace BSN.Commons.Infrastructure
+{
+    /// <summary>
+    /// Add Dynamic filterable query ability to <see cref="IRepository{T}"/> pattern"
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+	public interface IDynamicFilterableRepository<T> : IRepository<T> where T : class
+	{
+        /// <summary>
+        /// Get list of objects based on <paramref name="filters"/>
+        /// </summary>
+        /// <remarks>
+        /// This method retrive all objects based on filters and apply pagination.
+        /// So to retrive all objects that matched with filters, you need iterate all pages.
+        /// </remarks>
+        /// <param name="filters">
+        ///  is a comma-delimited list of <c>{Name}{Operator}{Value}</c> where
+        /// <list type="bullet">
+        ///  <item>
+        ///      <description>
+        ///          <c>{Name}</c> is the name of a property with the Sieve attribute or the name of a custom filter method for <c>TEntity</c>
+        ///          <list type="bullet">
+        ///              <item>
+        ///                  <description>
+        ///                      You can also have multiple names (for OR logic) by enclosing them in brackets and using a pipe delimiter,
+        ///                      eg. <c> (LikeCount|CommentCount)>10 </c> asks if LikeCount or <c> CommentCount is >10 </c>
+        ///                  </description>
+        ///              </item>
+        ///          </list>
+        ///      </description>
+        ///  </item>
+        ///  <item>
+        ///      <description>
+        ///         <c>{Operator}</c> is one of the Operators
+        ///      </description>
+        ///  </item>
+        ///  <item>
+        ///      <description>
+        ///          <c>{Value}</c>` is the value to use for filtering
```

Comment:
thank u for expressive comments. 
a character of \` is extra in this line.

Reply:
I fixed it.